In [1]:
import numpy  as np

In [2]:
from proj1_helpers    import *
from implementations  import *
from preprocessing    import *
from cross_validation import *

In [3]:
DATA_TRAIN_PATH = '../data/train.csv'

In [4]:
# Get data from file
y, tx, ids = load_csv_data(DATA_TRAIN_PATH)

##### Cross validation

In [5]:
# Cross validation 5 folds over several combinations
train_least_squares(y, tx)

Precission 0.705176 obtained for MVT=0 CT=0.75 OT=3 AD=1
Precission 0.727172 obtained for MVT=0 CT=0.75 OT=3 AD=2
Precission 0.69962 obtained for MVT=0 CT=0.75 OT=3 AD=3
Precission 0.7053360000000001 obtained for MVT=0 CT=0.75 OT=4 AD=1
Precission 0.73256 obtained for MVT=0 CT=0.75 OT=4 AD=2
Precission 0.711352 obtained for MVT=0 CT=0.75 OT=4 AD=3
Precission 0.705568 obtained for MVT=0 CT=0.75 OT=5 AD=1
Precission 0.734128 obtained for MVT=0 CT=0.75 OT=5 AD=2
Precission 0.717568 obtained for MVT=0 CT=0.75 OT=5 AD=3
Precission 0.706168 obtained for MVT=0 CT=0.75 OT=6 AD=1
Precission 0.7347159999999999 obtained for MVT=0 CT=0.75 OT=6 AD=2
Precission 0.7250439999999999 obtained for MVT=0 CT=0.75 OT=6 AD=3
Precission 0.705928 obtained for MVT=0 CT=0.8 OT=3 AD=1
Precission 0.7276360000000001 obtained for MVT=0 CT=0.8 OT=3 AD=2
Precission 0.700656 obtained for MVT=0 CT=0.8 OT=3 AD=3
Precission 0.705952 obtained for MVT=0 CT=0.8 OT=4 AD=1
Precission 0.732104 obtained for MVT=0 CT=0.8 OT=4 AD=

Precission 0.5309799999999999 obtained for MVT=0.25 CT=1 OT=3 AD=3
Precission 0.6400239999999999 obtained for MVT=0.25 CT=1 OT=4 AD=1
Precission 0.5532519999999999 obtained for MVT=0.25 CT=1 OT=4 AD=2
Precission 0.4143 obtained for MVT=0.25 CT=1 OT=4 AD=3
Precission 0.6241160000000001 obtained for MVT=0.25 CT=1 OT=5 AD=1
Precission 0.412528 obtained for MVT=0.25 CT=1 OT=5 AD=2
Precission 0.40632 obtained for MVT=0.25 CT=1 OT=5 AD=3
Precission 0.609168 obtained for MVT=0.25 CT=1 OT=6 AD=1
Precission 0.47749600000000003 obtained for MVT=0.25 CT=1 OT=6 AD=2
Precission 0.4337079999999999 obtained for MVT=0.25 CT=1 OT=6 AD=3
Precission 0.7045919999999999 obtained for MVT=0.5 CT=0.75 OT=3 AD=1
Precission 0.73418 obtained for MVT=0.5 CT=0.75 OT=3 AD=2
Precission 0.7118399999999999 obtained for MVT=0.5 CT=0.75 OT=3 AD=3
Precission 0.7060919999999999 obtained for MVT=0.5 CT=0.75 OT=4 AD=1
Precission 0.738644 obtained for MVT=0.5 CT=0.75 OT=4 AD=2
Precission 0.7188960000000001 obtained for MVT=0

Precission 0.714948 obtained for MVT=0.75 CT=0.95 OT=4 AD=1
Precission 0.744836 obtained for MVT=0.75 CT=0.95 OT=4 AD=2
Precission 0.7148999999999999 obtained for MVT=0.75 CT=0.95 OT=4 AD=3
Precission 0.7160759999999999 obtained for MVT=0.75 CT=0.95 OT=5 AD=1
Precission 0.7496240000000001 obtained for MVT=0.75 CT=0.95 OT=5 AD=2
Precission 0.7291559999999999 obtained for MVT=0.75 CT=0.95 OT=5 AD=3
Precission 0.716808 obtained for MVT=0.75 CT=0.95 OT=6 AD=1
Precission 0.750316 obtained for MVT=0.75 CT=0.95 OT=6 AD=2
Precission 0.738652 obtained for MVT=0.75 CT=0.95 OT=6 AD=3
Precission 0.6471680000000001 obtained for MVT=0.75 CT=1 OT=3 AD=1


KeyboardInterrupt: 

In [5]:
# Cross validation 10 folds over the best candidates from cell before
train_least_squares(y, tx)

Precission 0.7491920000000001 obtained for MVT=1 CT=0.875 OT=6 AD=2
Precission 0.7489399999999999 obtained for MVT=1 CT=0.9 OT=6 AD=2
Precission 0.749728 obtained for MVT=1 CT=0.925 OT=6 AD=2
Precission 0.7503 obtained for MVT=1 CT=0.95 OT=6 AD=2
Precission 0.44643999999999995 obtained for MVT=1 CT=0.975 OT=6 AD=2


(0.7503, 1, 0.95, 2)

In [5]:
# Cross validation 10 folds varying both CT and OT
train_least_squares(y, tx)

Precission 0.7494200000000001 obtained for MVT=1 CT=0.925 OT=6 AD=2
Precission 0.7500479999999999 obtained for MVT=1 CT=0.925 OT=7 AD=2
Precission 0.7500600000000001 obtained for MVT=1 CT=0.925 OT=8 AD=2
Precission 0.750024 obtained for MVT=1 CT=0.95 OT=6 AD=2
Precission 0.7507199999999999 obtained for MVT=1 CT=0.95 OT=7 AD=2
Precission 0.7504200000000001 obtained for MVT=1 CT=0.95 OT=8 AD=2


(0.7507199999999999, 1, 0.95, 2)

In [5]:
# Cross validation 10 folds varying both CT and OT
train_least_squares(y, tx)

Precission 0.74252 obtained for MVT=0.5 CT=0.95 OT=6.5 AD=2
Precission 0.7418 obtained for MVT=0.5 CT=0.95 OT=7.5 AD=2
Precission 0.74144 obtained for MVT=0.5 CT=0.95 OT=8.5 AD=2
Precission 0.750652 obtained for MVT=1 CT=0.95 OT=6.5 AD=2
Precission 0.75044 obtained for MVT=1 CT=0.95 OT=7.5 AD=2
Precission 0.750856 obtained for MVT=1 CT=0.95 OT=8.5 AD=2


(0.750856, 1, 0.95, 2)

In [5]:
train_least_squares(y, tx)

Precission 0.7502920000000001 obtained for MVT=1 CT=0.95 OT=8.25 AD=2
Precission 0.7505120000000001 obtained for MVT=1 CT=0.95 OT=8.75 AD=2
Precission 0.7496079999999999 obtained for MVT=1 CT=0.95 OT=9.25 AD=2
Precission 0.749996 obtained for MVT=1 CT=0.95 OT=12 AD=2


(0.7505120000000001, 1, 0.95, 2)

#### Final least squares parameters search

In [5]:
train_least_squares(y, tx)

Precission 0.7894119999999999 obtained for MVT=1 CT=0.8 OT=6.5 AD=2
Precission 0.789448 obtained for MVT=1 CT=0.8 OT=7.5 AD=2
Precission 0.78958 obtained for MVT=1 CT=0.8 OT=8.5 AD=2
Precission 0.7891800000000001 obtained for MVT=1 CT=0.85 OT=6.5 AD=2
Precission 0.7898799999999999 obtained for MVT=1 CT=0.85 OT=7.5 AD=2
Precission 0.7901559999999999 obtained for MVT=1 CT=0.85 OT=8.5 AD=2
Precission 0.790104 obtained for MVT=1 CT=0.9 OT=6.5 AD=2
Precission 0.791188 obtained for MVT=1 CT=0.9 OT=7.5 AD=2
Precission 0.7914359999999999 obtained for MVT=1 CT=0.9 OT=8.5 AD=2
Precission 0.79068 obtained for MVT=1 CT=0.95 OT=6.5 AD=2
Precission 0.7908000000000002 obtained for MVT=1 CT=0.95 OT=7.5 AD=2
Precission 0.791596 obtained for MVT=1 CT=0.95 OT=8.5 AD=2


KeyboardInterrupt: 

In [5]:
train_least_squares(y, tx)

Precission 0.7675919999999999 obtained for MVT=1 CT=0.9 OT=3.5 AD=2
Precission 0.78318 obtained for MVT=1 CT=0.9 OT=4.5 AD=2
Precission 0.790628 obtained for MVT=1 CT=0.9 OT=8 AD=2
Precission 0.7913840000000001 obtained for MVT=1 CT=0.9 OT=9 AD=2
Precission 0.791228 obtained for MVT=1 CT=0.9 OT=10 AD=2
Precission 0.791412 obtained for MVT=1 CT=0.9 OT=11 AD=2
Precission 0.7565120000000001 obtained for MVT=1 CT=0.95 OT=3.5 AD=2
Precission 0.7812119999999999 obtained for MVT=1 CT=0.95 OT=4.5 AD=2
Precission 0.791188 obtained for MVT=1 CT=0.95 OT=8 AD=2
Precission 0.7915840000000001 obtained for MVT=1 CT=0.95 OT=9 AD=2
Precission 0.7916160000000001 obtained for MVT=1 CT=0.95 OT=10 AD=2
Precission 0.791384 obtained for MVT=1 CT=0.95 OT=11 AD=2


(0.7916160000000001, 1, 0.95, 10, 2)

In [5]:
train_least_squares(y, tx)

Precission 0.385308 obtained for MVT=1 CT=0.975 OT=9.5 AD=2
Precission 0.388308 obtained for MVT=1 CT=0.975 OT=9.5 AD=3


KeyboardInterrupt: 

In [5]:
train_least_squares(y, tx)

Precission 0.791524 obtained for MVT=1 CT=0.925 OT=9.5 AD=2
Precission 0.791628 obtained for MVT=1 CT=0.925 OT=10.5 AD=2
Precission 0.7915 obtained for MVT=1 CT=0.925 OT=11.5 AD=2
Precission 0.7912 obtained for MVT=1 CT=0.925 OT=14 AD=2


(0.791628, 1, 0.925, 10.5, 2)

In [5]:
train_least_squares(y, tx)

Precission 0.7969999999999999 obtained for MVT=1 CT=0.925 OT=10.5 AD=3
Precission 0.8045640000000001 obtained for MVT=1 CT=0.925 OT=10.5 AD=4
Precission 0.7858560000000001 obtained for MVT=1 CT=0.925 OT=10.5 AD=5


(0.8045640000000001, 1, 0.925, 10.5, 4)

In [5]:
train_least_squares(y, tx)

Precission 0.804064 obtained for MVT=1 CT=0.925 OT=10.5 AD=4


(0.804064, 1, 0.925, 10.5, 4)

In [5]:
train_least_squares(y, tx)

Precission 0.8022560000000001 obtained for MVT=1 CT=0.925 OT=9.5 AD=4
Precission 0.8045120000000001 obtained for MVT=1 CT=0.925 OT=11.5 AD=4


(0.8045120000000001, 1, 0.925, 11.5, 4)

In [5]:
train_least_squares(y, tx)

Precission 0.8040839999999999 obtained for MVT=1 CT=0.9 OT=10.5 AD=4
Precission 0.803796 obtained for MVT=1 CT=0.95 OT=10.5 AD=4


(0.8040839999999999, 1, 0.9, 10.5, 4)

In [5]:
train_least_squares(y, tx)

Precission 0.803472 obtained for MVT=1 CT=0.925 OT=12.5 AD=4
Precission 0.804204 obtained for MVT=1 CT=0.925 OT=13.5 AD=4


(0.804204, 1, 0.925, 13.5, 4)

In [5]:
train_least_squares(y, tx)

Precission 0.79762 obtained for MVT=1 CT=0.925 OT=11.5 AD=3
Precission 0.796604 obtained for MVT=1 CT=0.925 OT=11.5 AD=5
Precission 0.787728 obtained for MVT=1 CT=0.925 OT=11.5 AD=6


(0.79762, 1, 0.925, 11.5, 3)

### At this point we know that MVT=1 CT=0.925 OT=11 AD=4 is a local minimum

In [5]:
train_least_squares(y, tx)

Precission 0.80396 obtained for MVT=1 CT=0.925 OT=11 AD=4


(0.80396, 1, 0.925, 11, 4)